integration-tests/bats/checkout.bats

```diff
@@ -41,17 +41,19 @@ SQL
     [ "$status" -eq 0 ]
     [[ "$output" =~ "modified" ]] || false
 
-    # Making additional changes to main, should carry them to feature without any problem
+    # Making additional changes to main, should error out because the changes are different from the feature branch
     dolt sql -q "insert into test values (3)"
-    dolt checkout feature
+    run dolt checkout feature
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "working sets exist on both branches" ]] || false
 
     run dolt sql -q "select count(*) from test"
     [ "$status" -eq 0 ]
     [[ "$output" =~ "3" ]] || false
 
     run dolt status
     [ "$status" -eq 0 ]
-    [[ "$output" =~ "modified" ]] || false
+    [[ "$output" =~ "On branch main" ]] || false
```

Comment:
Line 46 (`dolt checkout feature`) will now start to fail, since there are working set changes on both branches. I can revert most of these changes, but only if we use `-f` to force the old behavior: `dolt checkout -f feature`.